Assets/Mirror/Runtime/NetworkWriter.cs

```diff
@@ -128,10 +129,55 @@ public ArraySegment<byte> ToArraySegment()
             return new ArraySegment<byte>(buffer, 0, length);
         }
 
-        public void WriteByte(byte value)
-        {
-            EnsureLength(position + 1);
-            buffer[position++] = value;
+        // WriteBlittable<T> from DOTSNET.
+        // this is extremely fast, but only works for blittable types.
+        //
+        // Benchmark:
+        //   WriteQuaternion x 100k, Macbook Pro 2015 @ 2.2Ghz, Unity 2018 LTS (debug mode)
+        //
+        //                | Median |  Min  |  Max  |  Avg  |  Std  | (ms)
+        //     before     |  30.35 | 29.86 | 48.99 | 32.54 |  4.93 |
+        //     blittable* |   5.69 |  5.52 | 27.51 |  7.78 |  5.65 |
+        //
+        //     * without IsBlittable check
+        //     => 4-6x faster!
+        //
+        //   WriteQuaternion x 100k, Macbook Pro 2015 @ 2.2Ghz, Unity 2020.1 (release mode)
+        //
+        //                | Median |  Min  |  Max  |  Avg  |  Std  | (ms)
+        //     before     |   9.41 |  8.90 | 23.02 | 10.72 |  3.07 |
+        //     blittable* |   1.48 |  1.40 | 16.03 |  2.60 |  2.71 |
+        //
+        //     * without IsBlittable check
+        //     => 6x faster!
```

Comment:
I dont think we need this benchmark in the comments, we can link to the issue we want to reference it.